### PR TITLE
Documentation + Submodules

### PR DESCRIPTION
**Pull request by paraacha**
- Adds an updated Architecture Document (please have a look at the figure in the updated document) 
- Adds an updated User-stories Document
- Adds a revamped README file (please have a look at it) 
- Adds a PHPSEcLib-based submodule, an OpenSSL-based submodule and a user-passwords based submodule.  

The purpose of this PR is to get the module in shape for a review by anyone.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,48 @@
-# Pubkey Encrypt
-Drupal 8 Module for adding Credentials-based Public Key Encryption support into the Field Encrypt module. The encryption mechanism is based on ownCloud's Data Encryption Model.
+# Pubkey Encrypt for Drupal 8
 
 [![Build Status](https://travis-ci.org/d8-contrib-modules/pubkey_encrypt.svg?branch=8.x)](https://travis-ci.org/d8-contrib-modules/pubkey_encrypt)
+
+Provides support for encrypting data with users login-credentials. The mechanism is based on ownCloud's Data Encryption Model.
```
Comment (colans) [inline README.md]: "users" requires an apostrophe at the end as it's possessive and plural.

> users'

File: pubkey_encrypt.info.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: Pubkey Encrypt
-description: Adds Credentials-based Public Key Encryption support.
+description: Provides support for encrypting data with users login-credentials.
```
Comment (colans) [inline pubkey_encrypt.info.yml]: Same issue with "users" here too.  Also, I don't think "login-credentials" should be hyphenated here and in various places in the README.  Can probably just remove the "-" everywhere.

File: pubkey_encrypt_phpseclib/src/Plugin/AsymmetricKeysGenerator/PHPSecLib.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['key_size'] = array(
+      '#type' => 'select',
+      '#title' => t('Key size in bits'),
+      '#options' => [
+        '1024' => '1024',
```
Comment (colans) [inline pubkey_encrypt_phpseclib/src/Plugin/AsymmetricKeysGenerator/PHPSecLib.php]: I thought we weren't going to provide the small key option, but maybe I'm remembering wrong.  Can keep it in, but maybe add a '#description' here saying it's not recommended or that bigger keys are better.  We should probably add that description anyway.

File: README.md [hunk 4]
```diff
@@ -1,4 +1,48 @@
...
+
+Pubkey Encrypt leverages the Key & Encrypt modules for integrating it's encryption mechanism into a Drupal website. Accordingly, **the module generates an Encryption Profile for each role in the website.** This allows an administrator to select the set of users whose login-credentials should be used during the encryption/decryption processes. For example, if an administrator choses the "Premium User Encryption Profile" while encrypting the data, then the login-credentials of all users from the "Premium User" role would get used. Consequently, only these users would then be able to decrypt the encrypted data.
+
+* For encrypting field data using this mechanism, Field Encrypt module should be used along with this one.
+* For encrypting files using this mechanism, Encrypted Files module should be used along with this one.
```
Comment (nerdstein) [inline README.md]: We renamed this to File Encrypt --- https://github.com/d8-contrib-modules/file_encrypt

Comment (nerdstein) [inline README.md]: Still in dev and being actively worked on

File: pubkey_encrypt.info.yml (hunk 2 above)
Comment (nerdstein) [inline pubkey_encrypt.info.yml]: Agreed

**Pull request merged**